src/mbgl/util/tile_coordinate.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef MBGL_UTIL_TILE_COORDINATE
+#define MBGL_UTIL_TILE_COORDINATE
+
+#include <mbgl/style/types.hpp>
+#include <mbgl/map/transform_state.hpp>
+
+
+namespace mbgl {
+
+class TransformState;
+
+// Has floating point x/y coordinates.
+// Used for computing the tiles that need to be visible in the viewport.
```

Comment:
I still don't really understand why this class exists, what exactly it represents, or what makes it distinct from the other coordinate classes. Can you flesh out this comment more?